src/main/scala/de/upb/cs/swt/delphi/instanceregistry/authorization/AuthProvider.scala

```diff
@@ -76,19 +78,37 @@ class AuthProvider(authDAO: AuthDAO) extends AppLogging {
     }
   }
 
-  def generateJwt(userName: String): String = {
-    val validFor: Long = Registry.configuration.authenticationValidFor
+  def generateJwt(userName: String): UserToken = {
     val user = authDAO.getUserWithUsername(userName)
+    getUserToken(user.get)
+  }
+
+  def generateJwtByUserId(userId: Long): UserToken ={
+    val user = authDAO.getUserWithId(userId)
+    getUserToken(user.get)
+  }
 
+  def getUserToken(user: DelphiUser): UserToken ={
+    val validFor: Long = Registry.configuration.authenticationValidFor
+    val refreshTokenValidFor: Long = Registry.configuration.refreshTokenValidFor
     val claim = JwtClaim()
       .issuedNow
       .expiresIn(validFor * 60)
       .startsNow
-      . + ("user_id", user.get.userName)
-      . + ("user_type", user.get.userType)
+      . + ("user_id", user.id.get)
```

Comment:
You need to convert the ```user.id.get``` into a string before you put it into the claim here. The reason for that is that in line 191 where we parse the JWT we try to parse ```user_id``` as ```String```. I tried to use the token i got from calling ```/authenticate``` and it failed because of ```spray.Json.JsNumber cannot be cast to spray.json.JsString``` in line 191. You should get the same error when you try to actually use a token after calling ```/authenticate```